pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -1062,3 +1044,231 @@ def _run_solver() -> CoherentResults:
+                sequence,
+                modulation=with_modulation,
+                extended_duration=sequence.get_duration(
+                    include_fall_time=with_modulation
+                ),
+            ),
+            sequence.register,
+            sequence.device,
+            sampling_rate,
+            config,
+            evaluation_times,
+        )
+
+
+class Simulation:
+    r"""Simulation of a pulse sequence using QuTiP.
+
+    Args:
+        sequence: An instance of a Pulser Sequence that we
+            want to simulate.
+        sampling_rate: The fraction of samples that we wish to
+            extract from the pulse sequence to simulate. Has to be a
+            value between 0.05 and 1.0.
+        config: Configuration to be used for this simulation.
+        evaluation_times: Choose between:
+
+            - "Full": The times are set to be the ones used to define the
+              Hamiltonian to the solver.
+
+            - "Minimal": The times are set to only include initial and final
+              times.
+
+            - An ArrayLike object of times in µs if you wish to only include
+              those specific times.
+
+            - A float to act as a sampling rate for the resulting state.
+        with_modulation: Whether to simulated the sequence with the programmed
+            input or the expected output.
+    """
+
+    def __init__(
+        self,
+        sequence: Sequence,
+        sampling_rate: float = 1.0,
+        config: Optional[SimConfig] = None,
+        evaluation_times: Union[float, str, ArrayLike] = "Full",
+        with_modulation: bool = False,
+    ) -> None:
+        """Instantiates a Simulation object."""
+        self._seq = sequence
+        self._modulated = with_modulation
+        self._emulator = QutipEmulator.from_sequence(
+            self._seq, sampling_rate, config, evaluation_times, self._modulated
+        )
+
+    @property
+    def evaluation_times(self) -> np.ndarray:
+        """The times at which the results of this simulation are returned.
+
+        Args:
+            value: Choose between:
+
+                - "Full": The times are set to be the ones used to define the
+                  Hamiltonian to the solver.
+
+                - "Minimal": The times are set to only include initial and
+                  final times.
+
+                - An ArrayLike object of times in µs if you wish to only
+                  include those specific times.
+
+                - A float to act as a sampling rate for the resulting state.
+        """
+        return self._emulator.evaluation_times
+
+    @evaluation_times.setter
+    def evaluation_times(self, value: Union[str, ArrayLike, float]) -> None:
+        """Sets times at which the results of this simulation are returned."""
+        warnings.warn(
+            DeprecationWarning(
+                "Setting `evaluation_times` is deprecated,"
+                " use `set_evaluation_times` instead."
+            ),
```

Comment:
The `DeprecationWarnings` are not shown by default, it's better to change the warnings filter inside a context manager:
```suggestion
        with warnings.catch_warnings():
            warnings.simplefilter("always")
            warnings.warn(
                DeprecationWarning(
                    "Setting `evaluation_times` is deprecated,"
                    " use `set_evaluation_times` instead."
                ),
```